src/platform/P6/ConfigurationManagerImpl.cpp

```diff
@@ -0,0 +1,131 @@
+/*
+ *
+ *    Copyright (c) 2021 Project CHIP Authors
+ *    Copyright (c) 2018 Nest Labs, Inc.
+ *    All rights reserved.
+ *
+ *    Licensed under the Apache License, Version 2.0 (the "License");
+ *    you may not use this file except in compliance with the License.
+ *    You may obtain a copy of the License at
+ *
+ *        http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *    Unless required by applicable law or agreed to in writing, software
+ *    distributed under the License is distributed on an "AS IS" BASIS,
+ *    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *    See the License for the specific language governing permissions and
+ *    limitations under the License.
+ */
+
+/**
+ *    @file
+ *          Provides the implementation of the Device Layer ConfigurationManager object
+ *          for the PSoC6.
+ */
+/* this file behaves like a config.h, comes first */
+#include <platform/internal/CHIPDeviceLayerInternal.h>
+
+#include <platform/internal/GenericConfigurationManagerImpl.cpp>
+
+#include <platform/ConfigurationManager.h>
+#include <platform/KeyValueStoreManager.h>
+#include <platform/P6/P6Config.h>
+
+namespace chip {
+namespace DeviceLayer {
+
+using namespace ::chip::DeviceLayer::Internal;
+
+/** Singleton instance of the ConfigurationManager implementation object for the PSoC6.
+ */
+ConfigurationManagerImpl ConfigurationManagerImpl::sInstance;
+
+CHIP_ERROR ConfigurationManagerImpl::_Init()
+{
+#if CHIP_DEVICE_CONFIG_ENABLE_FACTORY_PROVISIONING
+#endif // CHIP_DEVICE_CONFIG_ENABLE_FACTORY_PROVISIONING
+
+    CHIP_ERROR err = CHIP_NO_ERROR;
+    bool failSafeArmed;
+
+    // Initialize the generic implementation base class.
+    err = Internal::GenericConfigurationManagerImpl<ConfigurationManagerImpl>::_Init();
+    SuccessOrExit(err);
+
+    // If the fail-safe was armed when the device last shutdown, initiate a factory reset.
+    if (_GetFailSafeArmed(failSafeArmed) == CHIP_NO_ERROR && failSafeArmed)
+    {
+        ChipLogProgress(DeviceLayer, "Detected fail-safe armed on reboot; initiating factory reset");
+        _InitiateFactoryReset();
+    }
+
+exit:
```

Comment:
this exit label has no cleanup code. Please convert file to use return/return macros instead of goto exit.

Reply:
Updated to use VerifyOrReturnError